### PR TITLE
Emit share with user type

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Enhancement: Expose user type of share receiver in share api

## Related Issue

- Part of: https://github.com/owncloud/web/issues/5634

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation ticket raised: https://github.com/owncloud/docs-server/issues/272
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: apps/files_sharing/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -99,7 +100,8 @@ public function __construct(array $urlParams = []) {
 				$c->query(NotificationPublisher::class),
 				$server->getEventDispatcher(),
 				$c->query(SharingBlacklist::class),
-				$c->query(SharingAllowlist::class)
+				$c->query(SharingAllowlist::class),
+				new UserTypeHelper(),
```
Comment (kulmann) [inline apps/files_sharing/lib/AppInfo/Application.php:104]: Is it not possible to get the `UserTypeHelper` via DI? 🤔

Comment (JanAckermann) [reply apps/files_sharing/lib/AppInfo/Application.php:104]: unfortunately no

Comment (jvillafanez) [reply apps/files_sharing/lib/AppInfo/Application.php:104]: Same question from me. Ideally it should be injected via DI, if that's not possible, a code comment can help to know the reason.
Having multiple instances isn't critical in this particular case, but I don't think it's expected under normal conditions.

**Review by kulmann — APPROVED** [review]

**Review by jvillafanez — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]
LGTM - I added `share_with_user_type` to 2 acceptance test scenarios to make sure that it really shows up.

**Comment by jnweiger** [comment]
Confirmed fixed in 10.10.0 RC2

/ocs/v1.php/apps/files_sharing/api/v1/shares?format=json has

* normal users shares: share_type=0, share_with_user_type=0
* guest users shares: share_type=0, share_with_user_type=1
* group shares: share_type=1 and no element share_with_user_type

share_with_user_type_string was not added.

**Pull request merged**